### ISSUE TITLE
Avaibality of the Visual Viewport API in non-secure contexts

### ISSUE THREAD

**Issue by tpodder** [bokand/ViewportAPI#57]
The spec does not mention if the Visual Viewport API is restricted to secure contexts only. Currently, in Chrome the Visual Viewport API is available in non-secure contexts. The W3C spec for secure contexts https://w3c.github.io/webappsec-secure-contexts/#new recommends all new features to be restricted to secure contexts. For existing features which are not widely used but widely implemented, the spec suggests applying this restriction and modifying the specs of the existing feature. What do you think about restricting the Visual Viewport API to secure contexts?

**Comment by bokand** [comment]
The link you provide specifically mentions _sensitive_ APIs which I would assume are those that can have privacy or other adverse consequences. I would assume VisualViewport is more akin to basic geometry information about the page (e.g. like window.innerWidth/Height) so I don't really see a reason it would need to be restricted - did this come up in a particular use case or example?

**Comment by theres-waldo** [comment]
I think we overlooked the fact that the linked policy only applies to sensitive APIs (which I agree Visual Viewport is not). I think part of the confusion is that for *new* APIs, we are trying to [restrict everything](https://blog.mozilla.org/security/2018/01/15/secure-contexts-everywhere/) (sensitive or not) to secure contexts.

**Comment by bokand** [comment]
Yeah, that sounds fine to me. Unfortunately, we've been shipping it like this for a while so there's a non-trivial number of uses - I'll have to do some investigation to see if that's practical to do.

**Comment by karlcow** [comment]
s/Avaibality/Availability/ in title.
